snapcraft/cli/_errors.py

```diff
@@ -0,0 +1,61 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import contextlib
+import functools
+import sys
+
+from ._options import get_project_options
+from . import echo
+
+import snapcraft.internal.errors
+
+
+@contextlib.contextmanager
+def exception_catcher(debug):
+    try:
+        yield
+    except snapcraft.internal.errors.SnapcraftError as e:
+        if debug:
+            raise
+        else:
+            echo.error(str(e))
+            sys.exit(1)
```

Comment:
Excellent idea. However, since the traceback ends up exiting 1, I defaulted to 2. That way we can at least distinguish between non-snapcraft errors and snapcraft errors. We can further flesh this out by reorganizing our errors a bit better.